### ISSUE TITLE
ViteExpress.static() not working

### ISSUE THREAD

**Issue by ReaganBruce** [szymmis/vite-express#51]
I have error-handling middleware that is blocking client side routes. Reading the documentation, I tried using ViteExpress.static() before the middleware; however it is not a value on the object. What file would you recommend serving to the client side?

Middleware: 
![image](https://github.com/szymmis/vite-express/assets/77300113/4ae0300b-0393-4479-847c-5e945607f7c3)

Server:
![image](https://github.com/szymmis/vite-express/assets/77300113/17c8c048-4139-43a4-84e7-7e45efc3c00c)

Thank you!

**Comment by xb36** [comment]
Hi @ReaganBruce. It is likely that you need to upgrade your version of vite-express, as `ViteExpress.static()` middleware feature was added just yesterday (see [#50](https://github.com/szymmis/vite-express/pull/50))

run `npm install vite-express --save` and you should be able to use the `.static()` middleware. Let us know if it works for you :)

**Comment by ReaganBruce** [comment]
Gotcha, thank you @xb36! Apologies, didn't realize it had been implemented yesterday

**Comment by ReaganBruce** [comment]
if you don't mind, I'd like to ask another question!

So, I have middleware that's blocking any 404 request, and on the client side it properly gives a "cannot get" message when going to "/" or "/test" for example.

Does ViteExpress.static() serve the HTML that is being blocked by the middleware? Apologies if I'm not incredibly clear.

**Comment by xb36** [comment]
The idea behind implementing ViteExpress.static() was to give you the choice of order. 
If you express.use(your_404_middleware) before ViteExpress.static(), you should be fine.
Order matters. 

Basically, using ViteExpress.static() should have the same effect as using express.static() in regards to your question. Actually, in production mode it literally does:
```js
//vite-express/src/main.ts
...
return express.static(distPath);
```

For further information on that topic have a look at [using express middleware](http://expressjs.com/en/guide/using-middleware.html) and [this StackOverflow answer](https://stackoverflow.com/a/33245894/7265288).

**Comment by szymmis** [comment]
Thanks a lot @xb36 for helping out here almost instantly and answering the questions!

Hi @ReaganBruce, really good to hear that you was able to resolve the problem just by updating the library.
When it comes to your question:
> Does ViteExpress.static() serve the HTML that is being blocked by the middleware?

The answer is no. The middleware that serves HTML is a separate one, injected at the end of the middleware stack no matter the order of your `app.use` calls. In sends index.html on every request as the route regex for that is "*". The idea is that we want to make Single Page Applications possible, but as in your case you can have some kind of middleware that knows what front-end routes are viable and sends 404 when needed.

As @xb36 said, the order of the `ViteExpress.static()` middleware matters, but in terms of static files e.g. JS, CSS, images etc. But when you request /some/page, it is a different middleware that handles it. The idea why you have options to use the `static` method by yourself but not the other middleware is that, when you have authorization you can easily check if the route equals let's say **/login**, and just let the request go because you want that route to be visible. But what about all those static files that are served also? That's why you just need to use `ViteExpress.static()` before that middleware and everything works because now static files are served no matter what so they aren't blocked when you want to display login page. 

It is also worth noting that the static files will be served no matter if you use `ViteExpress.static()` or not. You have that option just to explicitly specify the order that you want to have as @xb36 pointed out.

If you have more questions or something isn't clear yet, feel free to ask!

**Comment by ReaganBruce** [comment]
Thank you both for your responses!

I am adding the "ViteExpress.static()" to the beginning of both my error handler middlewares and am still getting a 404 when accessing any client side routing:
![image](https://github.com/szymmis/vite-express/assets/77300113/3cda407d-5d0f-4936-ac0f-dbf0453f5ce1)

I am fairly new to Vite, but I feel like I understand some of the nuances of it. Typically, I am use to serving index.html files from Express so the client side can read the routes. Example:
![image](https://github.com/szymmis/vite-express/assets/77300113/1065db72-e57c-4f07-a76a-61d198896dc0)

Since ViteExpress is handling on serving the files for me, I am a little confused on if I am doing something completely wrong, or if I should still try and serve the index.html file in the root of the project myself.

I understand both of you are extremely busy, and this is out of scope, but here is my repo in case either of you have a chance to look at it and point any glaring mistakes on my part: https://github.com/ReaganBruce/MERN_Portfolio

Thank you for being patient and helping!
